src/ln/channel.rs

```diff
@@ -412,7 +412,7 @@ impl Channel {
 	}
 
 	// Constructors:
-	pub fn new_outbound(fee_estimator: &FeeEstimator, keys_provider: &Arc<KeysInterface>, their_node_id: PublicKey, channel_value_satoshis: u64, push_msat: u64, user_id: u64, logger: Arc<Logger>, config: &UserConfig) -> Result<Channel, APIError> {
+	pub fn new_outbound(fee_estimator: &FeeEstimator, keys_provider: &Arc<KeysInterface>, their_node_id: PublicKey,channel_id: [u8; 32], channel_value_satoshis: u64, push_msat: u64, user_id: u64, logger: Arc<Logger>, config: &UserConfig) -> Result<Channel, APIError> {
```

Comment:
Instead of passing the 32 bytes down through the call stack here, maybe hook the rng and change the value returned in testing? In general I think we should be moving away from using the thread_rng/rand crate directly as we are trying to design for unknown environment. We already moved away from it a chunk with KeysInterface and further with #260, and this is actually the last place that rand_u832 is called, we should be able to trivially just move the current fuzztarget implementation into fuzz/fuzz_targets/full_stack_target and replace it with KeysInterface::get_session_key (maybe just renaming it to provide a random [u8; 32] and then having a default fn that converts it to a SecretKey).

Reply:
Trying to hook the rng to create a duplicate channel id was very difficult as the rng used cannot be reseeded without having the original handle. I tried many different ways and currently the only way I could do it was by using the rng of the fuzztarget feature. Unfortunately, this means the test will only run when you are testing with the fuzztarget feature enabled.

Secondly, I did attempted to move the fuzztarget rng into the KeysInterface of the full_stack_target but currently the chain::keysinterface is used in channel.rs and not the full_stack_target version. I might be missing something completely, but I'm not sure how to proceed? Please let me know what I should change.

Reply:
Oops, I wasnt clear, I was suggesting something more like #260 where you remove the RNG call and instead call a keysinterface function which provides 32 random bytes.

Reply:
That way you can hook the specific call in a test keys interface, without having any special mode for it.

Reply:
I changed it to be similar to #260, the channel id is now generated in the keysinterface.